src/arch/xtensa/boot_loader.c

```diff
@@ -93,6 +115,34 @@ static void parse_module(struct sof_man_fw_header *hdr,
 #define MAN_SKIP_ENTRIES 1
 #endif
 
+static uint32_t  get_fw_size_in_use(void)
+{
+	struct sof_man_fw_desc *desc =
+		(struct sof_man_fw_desc *)MANIFEST_BASE;
+	struct sof_man_fw_header *hdr = &desc->header;
+	struct sof_man_module *mod;
+	uint32_t fw_size_in_use = 0xffffffff;
+	int i;
+
+	/* Calculate fw size passed in BASEFW module in MANIFEST */
+	for (i = MAN_SKIP_ENTRIES; i < hdr->num_module_entries; i++) {
+		platform_trace_point(TRACE_BOOT_LDR_PARSE_MODULE + i);
+		mod = (void *)desc + SOF_MAN_MODULE_OFFSET(i);
+		if (strcmp((char *)mod->name, "BASEFW"))
+			continue;
+		for (i = 0; i < 3; i++) {
```

Comment:
Please replace 3 with some defined value. It's hard to tell what does it mean exactly.